client/verta/verta/_cli/deployment/update.py

```diff
@@ -51,10 +51,19 @@ def update_endpoint(path, run_id, strategy, canary_rule, canary_interval, canary
         raise click.BadParameter("experiment run with id {} not found".format(run_id))
 
     if strategy == 'direct':
-        endpoint.update(run, DirectUpdateStrategy())
+        strategy_obj = DirectUpdateStrategy()
     else:
         # strategy is canary
         strategy_obj = CanaryUpdateStrategy(canary_interval, canary_step)
         for rule in canary_rule:
             strategy_obj.add_rule(_UpdateRule._from_dict(json.loads(rule)))
-        endpoint.update(run, strategy_obj)
+
+    if env_vars:
+        env_vars_dict = json.loads(env_vars)
+    else:
+        env_vars_dict = None
+    try:
+        endpoint.update(run, strategy_obj, env_vars=env_vars_dict)
+    except TypeError as ex:
+        raise click.BadParameter(str(ex))
```

Comment:
At this point, it's okay to just have Python raise the full stack trace instead of wrapping in `BadParameter`. I recommend only using `BadParameter` for simple cases where the user's input to the CLI is wrong.
```suggestion

    endpoint.update(run, strategy_obj, env_vars=env_vars_dict)
```